app/src/androidTest/java/com/github/se/project/ui/components/SubjectsSelectorTest.kt

```diff
@@ -0,0 +1,44 @@
+package com.github.se.project.ui.components
+
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.createComposeRule
+import com.github.se.project.model.profile.Subject
+import org.junit.Assert.*
+import org.junit.Rule
+import org.junit.Test
+
+class SubjectsSelectorTest {
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Test
+  fun testSubjectsSelector_onClickDisplaysDropdownAndSelectsItem() {
+    val selectedSubjects =
+        mutableStateListOf<Subject>() // Utilise mutableStateListOf pour une réactivité fiable
+
```

Comment:
Small improvement could be to write the comment in English for better unanimity.